### PR TITLE
Add initial architecture docs

### PR DESCRIPTION
**Pull request by atmoos**
Add first simplistic architecture document. It describes

- The core design principles
- High level design
- renamed "Other" to "INoSystem" and "NonStandard"

### CODE REVIEW

File: architecture/Architecture.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+Every...
+
+- quantity is a struct
+- quantity has an _internal_ [unit](../quantities/units/IUnit.cs) that is either
+  - [SI](../quantities/units/Si/ISiUnit.cs) based
+  - [Imperial](../quantities/units/Imperial/IImperial.cs)
+  - [other](../quantities/units/Other/IOther.cs)
+- SI quantity supports [prefixing](../quantities/prefixes/IPrefix.cs) with
+  - a decimal prefix
+  - a binary prefix
```
Comment (RaduTerec) [inline architecture/Architecture.md:14]: What is a binary prefix?

File: architecture/Architecture.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+## Public API
+
+This library was built around an API I had in mind. The API should rely heavily on generics, which would make it very easy to use and extend for an arbitrary amount of units and (metric) prefixes.
+
+The minimal API for creation looks like this:
+
+```csharp
+public struct TQuantity : IMyDimension
+{
+    public static TQuantity Create<TPrefix, TUnit>(in Double value)
```
Comment (RaduTerec) [inline architecture/Architecture.md:40]: I have to read more about [`Contravariant`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/in-generic-modifier) before I can fully understand this concept.
I never used a doulbe `Where`. Cool stuff.

**Review by RaduTerec — DISMISSED** [review]

File: quantities/units/Imperial/Volume/Gallon.cs [hunk 3]
```diff
@@ -13,7 +13,7 @@ namespace Quantities.Units.Imperial.Volume;
     static T IInjectUnit<ILength>.Inject<T>(in Creator<ILength, T> inject, in Double self)
     {
         const Double galToCuFt = ToCuMetre / (Foot.ToMetre * Foot.ToMetre * Foot.ToMetre);
-        return inject.Other<Foot>(galToCuFt * self);
+        return inject.Imperial<Foot>(galToCuFt * self);
```
Comment (RaduTerec) [inline quantities/units/Imperial/Volume/Gallon.cs:16]: Could you rename `galToCuFt` to _galleonToCubicFeet_?

**Review by RaduTerec — DISMISSED** [review]
Looks good.

I suggested a few renamings. We can talk about the bigger stuff in a different review.

File: quantities/measures/Measures.cs [hunk 4]
```diff
@@ -35,8 +36,15 @@ namespace Quantities.Measures;
     public static Double FromSi(in Double value) => TPrefix.FromSi(TUnit.FromSi(in value));
```
[the remark this answers is not in the document]
Comment (atmoos) [reply quantities/measures/Measures.cs:36]: You're absolutely right. I opened issue No #4 to resolve this.

One note though: There is a priori no need for any relationships between any of these measures. The only thing relevant is that the implement some "Measure" interface.

But that may be the overall problem here. Perhaps we should give these measures some relationship with one another...

File: quantities.test/LengthTest.cs [hunk 5]
```diff
@@ -163,7 +163,7 @@ public void SiLengthByOtherTimeIsVelocity()
         actual.Matches(expected);
     }
     [Fact]
-    public void OtherLengthByTimeIsVelocity()
+    public void ImperialLengthByTimeIsVelocity()
     {
         Length distance = Length.Imperial<Mile>(70);
         Time duration = Time.In<Hour>(2);
```
[the remark this answers is not in the document]
Comment (atmoos) [reply quantities.test/LengthTest.cs:169]: Implemented

File: quantities/units/Imperial/Volume/FluidOunce.cs [hunk 6]
```diff
@@ -13,7 +13,7 @@ namespace Quantities.Units.Imperial.Volume;
     static T IInjectUnit<ILength>.Inject<T>(in Creator<ILength, T> inject, in Double self)
     {
         const Double flOzToCuIn = ToCuMeter / (Inch.ToMetre * Inch.ToMetre * Inch.ToMetre);
```
[the remark this answers is not in the document]
Comment (atmoos) [reply quantities/units/Imperial/Volume/FluidOunce.cs:15]: Implemented

File: quantities/units/Imperial/Volume/Gallon.cs (hunk 3 above)
Comment (atmoos) [reply quantities/units/Imperial/Volume/Gallon.cs:16]: Implemented

**Pull request merged**